gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -587,9 +610,52 @@ MM_MemoryPoolAddressOrderedList::collectorAllocate(MM_EnvironmentBase *env, MM_A
 	return addr;
 }
 
+/* Check the free entry being used and align it to appropriate boundaries for TLH allocation as required by GC.
+ * SATB Sweep has strict requirements for TLH to not span multiple Sweep Chunks. */
+bool
+MM_MemoryPoolAddressOrderedList::alignTLHForGC(MM_EnvironmentBase *env,  MM_HeapLinkedFreeHeader *freeEntry, uintptr_t *consumedSize)
+{
+	MM_GCExtensionsBase *extensions = env->getExtensions();
```

Comment:
if you keep alignment in _tlhAlignmentSize, then you can get rid of extension and env.